master/internal/core.go

```diff
@@ -966,10 +966,15 @@ func (m *Master) Run(ctx context.Context, gRPCLogInitDone chan struct{}) error {
 	// Other static files should only be cached for a short period of time
 	cacheFileShortTerm := regexp.MustCompile(`.(antd.\S+(.css)|ico|png|jpe*g|gif|svg)$`)
 
+	// API endpoints
+	apiRegex := regexp.MustCompile(`^/api/.+$`)
+
 	gzipConfig := middleware.GzipConfig{
 		Skipper: func(c echo.Context) bool {
```

Comment:
question: is there something we don't wanna compress? I followed the current code, but not sure if we want `Skipper` here

Reply:
hmmm I'm not familiar with compression... @maxrussell would you have any thoughts?

Reply:
Nothing I can think of. Seems like a good change.

Reply:
ok, i'll remove `Skipper` and relavant variables, then merge this PR. thanks for the review

Reply:
After some investigation, i conclude that `Skipper` is still required for gzip.

i found that `test_gc_checkpoints_lfs` test case should avoid being compressed on CI (look at [CI error](https://app.circleci.com/pipelines/github/determined-ai/determined/50208/workflows/5e7dd7b5-a455-4277-a575-beab314c49ee/jobs/2207487/parallel-runs/3))

Also, according to @azhou-determined, 
> gzip didn’t play nicely with the tensorboard profiler plugin. Tensorboard wouldn’t load when we compressed the responses